specs/master/cluster-init/files/host_provider/src/cyclecloud_provider.py

```diff
@@ -1189,6 +1188,29 @@ def periodic_cleanup(self):
             logger.exception("Could not cleanup old requests")
 
 
+def bucket_priority(nodearrays, bucket_nodearray, b_index):
+    prio = bucket_nodearray.get("Priority")
+    if isinstance(prio, str):
+        try:
+            prio = int(float(prio))
+        except Exception:
+            prio = None
+    if isinstance(prio, float):
+        prio = int(prio)
+    if isinstance(prio, int):
+        if prio < 0:
+            prio = None
+    if prio is not None and not isinstance(prio, int):
+        prio = None    
+    if prio is None:
+        nodearray_names = [x["name"] for x in reversed(nodearrays)]
+        prio = (nodearray_names.index(bucket_nodearray["name"]) + 1) * 10
+    if prio > 0:
+        return prio * 1000 - b_index
+    assert prio == 0
```

Comment:
Is assert the right choice here?
Let's at least throw an exception with meaningful error (just with a message that clearly states "This should NEVER happen"
ex. assert prio == 0, (
        f'Unexpected prio {prio} - should ALWAYS be >= 0.')